### PR TITLE
upgrade aws provider

### PR DESCRIPTION
**Pull request by AlyRagab**
- Upgrading the aws provider version to be "~> 3.0" because the current "2.0" does not work if we have another resources depend on version 3.0

- Error : 

Initializing provider plugins...
- terraform.io/builtin/terraform is built in to Terraform
- Finding hashicorp/null versions matching "~> 2.0"...
- Finding hashicorp/aws versions matching "~> 3.0, ~> 2.0"...
- Using hashicorp/null v2.1.2 from the shared cache directory

Error: Failed to query available provider packages

Could not retrieve the list of available versions for provider hashicorp/aws:
no available releases match the given constraints ~> 3.0, ~> 2.0

### CODE REVIEW

**Comment by joe-niland** [comment]
Hi @AlyRagab 
We are using lower bound pinning going forward.

Could you please adjust this for Terraform and for the providers?

**Comment by joe-niland** [comment]
Here's an example https://github.com/cloudposse/terraform-aws-codebuild/pull/70

File: versions.tf [hunk 1]
```diff
@@ -2,7 +2,7 @@ terraform {
   required_version = ">= 0.12.0, < 0.14.0"
```
Comment (aknysh) [inline versions.tf:2]:
```suggestion
  required_version = ">= 0.12.0"
```

File: versions.tf [hunk 2]
```diff
@@ -2,7 +2,7 @@ terraform {
...
 
   required_providers {
-    aws  = "~> 2.0"
+    aws  = "~> 3.0"
```
Comment (aknysh) [inline versions.tf:5]:
```suggestion
    aws  = ">= 2.0"
```

File: versions.tf [hunk 3]
```diff
@@ -2,7 +2,7 @@ terraform {
...
     null = "~> 2.0"
```
Comment (aknysh) [inline versions.tf:6]:
```suggestion
    null = "=> 2.0"
```

**Review by aknysh — COMMENTED** [review]
thanks @AlyRagab , please see comments.
Please also run the following commands from the repo root:

```
make init
make github/init
make readme
```

**Comment by AlyRagab** [comment]
Thanks @aknysh for your comments

**Review by aknysh — CHANGES_REQUESTED** [review]
@AlyRagab 
Sorry, but we also need to update the test to use `Go` modules.
Can you please run these commands:

```
make init
make github/init
cd test/src/
go mod init github.com/cloudposse/$(cd ../../ && basename `pwd`)
go mod tidy
git rm Gopkg.*
git add go.*
```

Thanks a lot for your contribution.

**Pull request closed without merging**